### PR TITLE
Unit Test:  Helper/ApiTraitTest

### PR DESCRIPTION
**Pull request by YevhenShBolt**
# Description
Please include a summary of the change and which issue is fixed. Include the motivation for the changes, and comment on your PR if necessary for clarity.

Fixes: https://app.asana.com/0/1131034199022699/1157340263780225/f

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

\#changelog

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Test

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

**Comment by matt-thomason** [comment]
@YevhenShBolt Is this PR still under development? Do you have an Asana task to track this?

**Comment by YevhenShBolt** [comment]
@matt-thomason i can't find task for it. I finished development here.

**Comment by adentong** [comment]
@YevhenShBolt If development work is finished can you remove the WIP in the title? Let's try to get this merged.

**Comment by YevhenShBolt** [comment]
@adentong  sorry i missed title. Done.

**Comment by matt-thomason** [comment]
^ Echoing @adentong

**Comment by YevhenShBolt** [comment]
@matt-thomason @adentong please review again.

**Comment by YevhenShBolt** [comment]
CC:  @adentong

File: tests/unit/testsuite/Bolt/Boltpay/Helper/ApiTraitTest.php [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                    'object' => '',
+                    'type' => '',
+                    'storeId' => '',
+                    'publishable_key_multipage' => '',
+                    'api_key' => '',
+                    'context' => array(),
+                    'response' => array()
+                )
+            ),
+
```
Comment (adentong) [inline tests/unit/testsuite/Bolt/Boltpay/Helper/ApiTraitTest.php]: nit: Remove empty line

File: tests/unit/testsuite/Bolt/Boltpay/Helper/ApiTraitTest.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        $guzzle->method('post')->will($this->returnValue($response));
+        $guzzle->method('get')->will($this->returnValue($response));
+        $mock->method('getApiClient')->will($this->returnValue($guzzle));
+        $mock->method('getContextInfo')->will($this->returnValue($case['context']));
+        $mock->method('addMetaData');
+        // Start test
+        $result = $mock->transmit($case['command'], $case['data'], $case['object'], $case['type'], $case['storeId']);
+        $this->assertInternalType('array', $result);
+        $this->assertEquals($case['expect'], $result);
+
```
Comment (adentong) [inline tests/unit/testsuite/Bolt/Boltpay/Helper/ApiTraitTest.php]: nit: Remove empty line

File: tests/unit/testsuite/Bolt/Boltpay/Helper/ApiTraitTest.php [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                    'publishable_key_multipage' => '',
+                    'api_key' => '',
+                    'context' => array(),
+                    'response' => null,
+                    'exception' => 'Test Rejected request'
+                )
+            ),
+        );
+    }
+
```
Comment (adentong) [inline tests/unit/testsuite/Bolt/Boltpay/Helper/ApiTraitTest.php]: nit: Remove empty line

File: app/code/community/Bolt/Boltpay/Helper/ApiTrait.php [hunk 4]
```diff
@@ -182,7 +182,6 @@ public function transmit($command, $data, $object='merchant', $type='transaction
         }
     }
 
-
```
Comment (adentong) [inline app/code/community/Bolt/Boltpay/Helper/ApiTrait.php]: I think the isResponseError function is unused? If it is then please remove it.

**Comment by adentong** [comment]
@YevhenShBolt

**Comment by YevhenShBolt** [comment]
@adentong i updated code.
several notes:
- Github and Magento require a new line at the end of the file.
Screenshot 2020-02-03 at 20 09 12
- >Can you also add tests for verify_hook?

Yes, i am going to create a new PR for it. I see another test file `app/code/community/Bolt/Boltpay/Helper/ApiTrait.php` which require a refactoring and merge with current test file, and also i will include new tests for `verify_webhook`.

If you ok with it please approve or let me know your opinion.

**Comment by adentong** [comment]
@YevhenShBolt Thanks. Didn't know about newline at the end of file being required.

**Comment by adentong** [comment]
@matt-thomason @ryan-timothy-albert @mccottry Could one of you also take a look at this?

**Pull request merged**